modules/imap/functions.php

```diff
@@ -1425,3 +1425,50 @@ function parse_sieve_config_host($host) {
     // $host = '$scheme://'.$host;
     return [$host, $port, $tls];
 }}
+
+if (!hm_exists('connect_to_imap_server')) {
+    function connect_to_imap_server($address, $name, $port, $user, $pass, $tls, $imap_sieve_host, $enableSieve, $type, $context, $hidden = false) {
+        Hm_IMAP_List::init($context->user_config, $context->session);
```

Comment:
This is really needed in only one place and is executed in Hm_Handler_load_imap_servers_from_config - if you need the list initialized from the repo, just add that handler module to the page you are going to use these.